TUTORIAL.md

```diff
@@ -0,0 +1,205 @@
+            &data,
+        )
+    }
+}
+
+// API test skeleton
+
+#[test]
+fn my_api_test() {
+    let mut testkit = TestKitBuilder::validator()
+        .with_service(MyService::new())
+        .create();
+    fill_storage_with_data(&mut testkit);
+    // Check API responses
+    let api = testkit.api();
+    assert_eq!(api.get_public_data(), ApiResponsePublicData::new(...));
+    ...
+}
+```
+
+## Advanced usage
+
+Here are examples of more complex and less common cases.
+
+### Oracles testing
+
+The oracle in this case is a service which can produce transactions with external data after the commit of the block,
+[`exonum-time`][exonum-time] and [`exonum-btc-anchoring`][exonum-btc-anchoring] are examples of these kinds of oracles.
+In this way, transactions created during the `handle_commit` execution will be stored in `TestKit` memory pool.
+
+```rust
+// Create testkit with the service which creates transaction with the height
+// of latest committed block after commit.
+let mut testkit = TestKitBuilder::validator()
+    .with_service(HandleCommitService)
+    .create();
+// Call the `handle_commit` event.
+testkit.create_block();
+// Check that `handle_commit` has been invoked at the correct height.
+let tx = TxAfterCommit::new_with_signature(Height(1), &Signature::zero());
+assert!(testkit.mempool().contains_key(&tx.hash()));
+```
+
+In order to invoke a `handle_commit` event, you must create a block.
+If the oracle has to fetch any data from external world, you must create a mock object that
+would generate said external data to accomplish testing.
+
+```rust
+// Provide a mock object for the service.
+let mut cruel_world = ExternalApiMock::new();
+let mut testkit = TestKitBuilder::validator()
+    .with_service(MyOracleService::with_client(cruel_world.client()))
+    .create();
+// Expect a request from the service.
+cruel_world.expect_api_call(ApiCallInfo { ... })
+    .with_response_ok(ApiResponse { ... });
+// Call the `handle_commit` event.
+testkit.create_block();
+let expected_tx = MyOracleTx::new(...);
+// Check that the expected transaction is in the memory pool.
+assert!(testkit.mempool().contains_key(&expected_tx.hash()));
+```
+
+### Configuration changes testing
+
+If your service has its own configuration, you may need to test the response to a configuration change.
+With the testkit you can create a configuration change proposal and commit it.
+
+```rust
+let mut testkit = TestKitBuilder::validator()
+    .with_service(MyOracleService::new())
+    .create();
+// Create a configuration change proposal.
+let proposal = {
+    let mut cfg = testkit.configuration_change_proposal();
+    cfg.set_actual_from(cfg_change_height);
+    cfg.set_service_config("my_service", MyServiceCfg { ... });
+    cfg
+};
+let stored = proposal.stored_configuration().clone();
+testkit.commit_configuration_change(proposal);
+// Check that the following configuration before block creation is none.
+use exonum::blockchain::Schema;
```

Comment:
Nitpick: I'd either move the `use` statement upper, or inserted a newline above the comment "Check that the following configuration..." to clearly separate this block logically.